src/AspNet.Security.OAuth.Alipay/AlipayAuthenticationHandler.cs

```diff
@@ -0,0 +1,227 @@
+                                /* Body: */ await response.Content.ReadAsStringAsync());
+
+                throw new HttpRequestException("An error occurred while retrieving user information.");
+            }
+
+            using var stream = await response.Content.ReadAsStreamAsync(Context.RequestAborted);
+            using var document = JsonDocument.Parse(stream);
+            var rootElement = document.RootElement;
+
+            if (!rootElement.TryGetProperty("alipay_user_info_share_response", out var mainElement))
+            {
+                string errorCode = rootElement.GetProperty("error_response").GetProperty("code").GetString() !;
+                throw new Exception($"An error (Code:{errorCode}) occurred while retrieving user information.");
+            }
+
+            if (!ValidateReturnCode(mainElement, out string code))
+            {
+                throw new Exception($"An error (Code:{code}) occurred while retrieving user information.");
+            }
+
+            identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, mainElement.GetString("user_id") !, ClaimValueTypes.String, Options.ClaimsIssuer));
+
+            var principal = new ClaimsPrincipal(identity);
+            var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, mainElement);
+
+            context.RunClaimActions();
+
+            await Options.Events.CreatingTicket(context);
+
+            return new AuthenticationTicket(context.Principal!, context.Properties, Scheme.Name);
+        }
+
+        protected override string FormatScope() => FormatScope(Options.Scope);
+
+        protected override string FormatScope(IEnumerable<string> scopes) => string.Join(',', Options.Scope);
+
+        /// <summary>
+        /// Check the code sent back by server for potential server errors.
+        /// </summary>
+        /// <param name="e">Main part of json document from response</param>
+        /// <param name="code">Returned code from server</param>
+        /// <remarks>See https://opendocs.alipay.com/open/common/105806 for details.</remarks>
+        /// <returns>True if succeed, otherwise false.</returns>
+        private static bool ValidateReturnCode(JsonElement e, out string code)
+        {
+            if (!e.TryGetProperty("code", out var codeElement))
+            {
+                code = string.Empty;
+                return true;
+            }
+
+            code = codeElement.GetString() !;
+            return code == "10000";
+        }
+
+        /// <summary>
+        /// Gets the RSA2(SHA256 with RSA) signature.
+        /// </summary>
+        /// <param name="sortedPairs">Sorted key-value pairs</param>
+        /// <param name="privateKeyBytes">Private key</param>
+        private static string GetRSA2Signature([NotNull] SortedDictionary<string, string> sortedPairs, byte[] privateKeyBytes)
+        {
+            var builder = new StringBuilder(128);
```

Comment:
```suggestion
        private static string GetRSA2Signature([NotNull] SortedDictionary<string, string> sortedPairs)
        {
            byte[] privateKeyBytes = Convert.FromBase64String(Options.ClientSecret);
            var builder = new StringBuilder(128);
```

Reply:
Just confirm. If so, `GetRSA2Signature` will be non static due to `Options`.

Reply:
Correct.